src/lib/io/sol-gpio-impl-riot.c

```diff
@@ -59,8 +63,26 @@ static void
 gpio_process_cb(void *data)
 {
     struct sol_gpio *gpio = data;
+    bool val;
 
-    gpio->irq.cb((void *)gpio->irq.data, gpio);
+    val = sol_gpio_read(gpio);
+    gpio->irq.cb((void *)gpio->irq.data, gpio, val);
```

Comment:
keep current behavior and document this last part you clarified in the docs? (i didn't check the docs, maybe its there already)